openpype/hosts/maya/plugins/publish/extract_arnold_scene_source.py

```diff
@@ -113,34 +111,63 @@ def process(self, instance):
         instance.data["representations"].append(representation)
 
     def _extract(self, nodes, attribute_data, kwargs):
-        self.log.info("Writing: " + kwargs["filename"])
+        self.log.info(
+            "Writing {} with:\n{}".format(kwargs["filename"], kwargs)
+        )
         filenames = []
         # Duplicating nodes so they are direct children of the world. This
         # makes the hierarchy of any exported ass file the same.
-        with delete_after() as delete_bin:
+        with lib.delete_after() as delete_bin:
             duplicate_nodes = []
             for node in nodes:
+                # Only interested in transforms:
+                if cmds.nodeType(node) != "transform":
+                    continue
+
+                # Only interested in transforms with shapes.
+                shapes = cmds.listRelatives(
+                    node, shapes=True, fullPath=True
+                ) or []
+                if not shapes:
+                    continue
+
+                parent = cmds.listRelatives(
+                    node, parent=True, fullPath=True
+                )[0]
                 duplicate_transform = cmds.duplicate(node)[0]
+                duplicate_transform = "{}|{}".format(
+                    parent, duplicate_transform
+                )
+
+                if cmds.listRelatives(duplicate_transform, parent=True):
```

Comment:
We're first prefixing with the parent name, but then checking whether it has a parent? Seems off?

ALso, should duplicate return the valid path? If not, then we should do `cmds.ls(duplicate_transform, long=True)[0]` instead because it'll otherwise still likely fail if the node name isn't unique. I'm pretty sure duplicate will then already be returning the shortest unique path.

Reply:
Yeah, have simplified this.